api/turing/service/router_deployment_service.go

```diff
@@ -351,7 +351,9 @@ func (ds *deploymentService) createServices(
 	}
 
 	// Ensembler
-	if routerVersion.Ensembler != nil && routerVersion.Ensembler.Type == models.EnsemblerDockerType {
+	if routerVersion.Ensembler != nil &&
+		((routerVersion.Ensembler.Type == models.EnsemblerPyFuncType && routerVersion.Ensembler.DockerConfig != nil) ||
```

Comment:
I see this check being used quite often and maybe there should be a utility function that checks if its either a EnsemblerPyFuncType or EnsemblerDockerType

Reply:
That works too, but I think given the discussion above where we're changing the logic of the check to verify if an ensembler has a docker config or not, I'm not sure if it's excessive to replace `if routerVersion.Ensembler != nil && routerVersion.Ensembler.DockerConfig != nil` with `if hasDockerConfig(routerVersion.Ensember)` :/

Reply:
Definitely not excessive, I actually prefer the latter because shows the intent of the person writing it. `hasDockerConfig()` has more meaning than `A && B`.